CHANGELOG.md

```diff
@@ -1,5 +1,213 @@
+
+However, you can **return to the previous behavior** by using the `:userProfile` browser option.
+
+```sh
+testcafe firefox:userProfile tests/test.js
+```
+
+```js
+runner
+    .src('tests/fixture1.js')
+    .browsers('firefox:userProfile')
+    .run();
+```
+
+#### :gear: Customizable timeout to wait for the `window.load` event ([#1645](https://github.com/DevExpress/testcafe/issues/1645))
+
+Previously, TestCafe started a test when the `DOMContentLoaded` event was raised. However, there are many pages that execute some kind of initialization code on the `window.load` event (which is raised after `DOMContentLoaded` because it waits for all stylesheets, images and subframes to load). In this instance, you need to wait for the `window.load` event to fire before running tests.
+
+With this release, TestCafe waits for the `window.load` event for `3` seconds.
+We have also added a `pageLoadTimeout` setting that allows you to customize this interval.
+You can set it to `0` to skip waiting for `window.load`.
+
+The following examples show how to use the `pageLoadTimeout` setting from the command line and API.
+
+```sh
+testcafe chrome test.js --page-load-timeout 0
+```
+
+```js
+runner.run({
+    pageLoadTimeout: 0
+});
+```
+
+You can also use the `setPageLoadTimeout` method in test API to set the timeout for an individual test.
+
+```js
+fixture `Page load timeout`
+    .page `http://devexpress.github.io/testcafe/example/`;
+
+test(`Page load timeout`, async t => {
+    await t
+        .setPageLoadTimeout(0)
+        .navigateTo('http://devexpress.github.io/testcafe/');
+});
+```
+
+#### :gear: Access messages output by the tested app to the browser console ([#1738](https://github.com/DevExpress/testcafe/issues/1738))
+
+Use the `t.getBrowserConsoleMessages` method to obtain messages that the tested app output to the browser console.
+
+This method returns the following object.
+
+```js
+{
+    error: ["Cannot access the 'db' database. Wrong credentials.", '...'], // error messages
+    warn: ['The setTimeout property is deprecated', '...'],                // warning messages
+    log: ['[09:12:08] Logged in', '[09:25:43] Changes saved', '...'],      // log messages
+    info: ['The application was updated since your last visit.', '...']    // info messages
+}
+```
+
+Note that this method returns only messages posted via the `console.error`, `console.warn`, `console.log` and `console.info` methods. Messages output by the browser (like when an unhandled exception occurs on the page) will not be returned.
+
+The following example shows how to check the React prop types for errors using the `t.getBrowserConsoleMessages` method.
+
+```js
+// check-prop-types.js
+import { t } from 'testcafe';
+
+export default async function () {
+    const { error } = await t.getBrowserConsoleMessages();
+
+    await t.expect(error[0]).notOk();
```

Comment:
For me better looks following check:
```
await t.expect(error.length).notOk();
```
Because strictly saying if `error[0]` will empty string this assertion will be ok and it's not what's expected.

Reply:
`error[0]` is better because you will see the error text in the report.

In your case you'll have a report like: `expected no errors, but it was n`

Reply:
I don't quite agree but it's not big deal

Reply:
Which error in CI do you prefer to see:
`A value expected to be 'undefined' but it was 'Prop Type Error: a specific type error is wrong'`
or
`It expected there are no errors on the page but it was 1 error`

Reply:
I expect no errors at all and if there are some than something went wrong and I need to debug. I agree `A value expected to be 'undefined' but it was 'Prop Type Error: a specific type error is wrong'` variant is much more informative but I just said about one border case: when e.g. `error -> ['', '','']`.
Any way I don't mind staying `await t.expect(error[0]).notOk();` :)